### DISCUSSION TITLE
How many parallel TCP connections are supported by DETA?

### DISCUSSION THREAD

**Discussion by jnsougata** [deta/cloud-docs#220] [Help] [answered]
....
Getting `aiohttp.client_exceptions.ServerDisconnectedError: Server disconnected` most probably due to parallel connection limits!

**Comment by abdelhai** [comment, marked as answer]
Hi, could you explain more how are you getting? Also what service?

**Reply by jnsougata** [reply]
service: `base`

```py
async def add_field(self, key: str, field: Field, force: bool = False) -> Dict[str, Any]:

        if not force:
            return await self.update(key=key, updates=[Update.set([field])])
        try:
            return await self.update(key=key, updates=[Update.set([field])])
        except NotFound:
            return await self.put(key=key, field=field)
```
here you can see I am trying to update an existing filed or putting a new field... 
it mostly works fine but sometimes it's gving me the `aiohttp.client_exceptions.ServerDisconnectedError: Server disconnected` very randomly...

**Reply by abdelhai** [reply]
seems like we're using the [aiohttp's default connections limit](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.BaseConnector) of `100`.

**Reply by jnsougata** [reply]
No I get it.. Thanks
